hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -204,6 +204,9 @@ public void startScanningForJobs() {
                             + jobResult);
                     return;
                 }
+                if (!config.isResourceUploadEnabled() && !jobConfig.getResourceConfigs().isEmpty()) {
+                    throw new JetException("Resource upload is disabled and JobConfig contains resources");
```

Comment:
This message could be improved with a clear guidance for the user to fix the issue:
E.g.
```
The JobConfig contains resources to upload, but the resource upload is disabled. Either remove the resources from the job config or set the property 'hazelcast.jet.resourceUploadEnabled' to true
```